### ISSUE TITLE
Windows/GCE: kubelet does not write logs to file properly

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#77416]
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce/1124330363452133383/artifacts/e2e-3e2da0347d-36623-windows-node-group-sv9r/kubelet.log

The log file (kubelet.log) contains the flags and nothing else. This is a regression on the master branch since 1.14 works fine.

/cc @pjh @mtaufen

**Comment by pjh** [comment]
Looks like this has been happening since at least 04-19, bummer. https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce/1119365887820828675

**Comment by pjh** [comment]
Last time we got a full kubelet.log: [1119000479460757508](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce/1119000479460757508/artifacts/e2e-1459f119e5-36623-windows-node-group-9d76/) on 04-18.

First time we got the short kubelet.log: [1119030928954363908](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce/1119030928954363908/artifacts/e2e-225bd045a1-36623-windows-node-group-d483/) on 04-19.

**Comment by yujuhong** [comment]
Suspects:
https://github.com/kubernetes/kubernetes/compare/0772f852add44eab917409c291a5d82974bea822...d308368b2d720f732c9bef6e72ef6131388beb04

My bet is on the klog update:
https://github.com/kubernetes/kubernetes/pull/76474

**Comment by yujuhong** [comment]
kube-proxy does not have this problem, but its log configuration is identical to kubelet's.

Configuration from kubelet.log:
```
I0503 02:27:55.877934    4344 flags.go:33] FLAG: --alsologtostderr="false"
I0503 02:27:55.877934    4344 flags.go:33] FLAG: --log-backtrace-at=":0"
I0503 02:27:55.877934    4344 flags.go:33] FLAG: --log-dir=""
I0503 02:27:55.877934    4344 flags.go:33] FLAG: --log-file="C:\\etc\\kubernetes\\logs\\kubelet.log"
I0503 02:27:55.877934    4344 flags.go:33] FLAG: --log-file-max-size="1800"
I0503 02:27:55.877934    4344 flags.go:33] FLAG: --log-flush-frequency="5s"
I0503 02:27:55.877934    4344 flags.go:33] FLAG: --logtostderr="false"
```

Configuration from kube-proxy.log:
```
I0506 23:59:59.377592    4080 flags.go:33] FLAG: --alsologtostderr="false"
I0506 23:59:59.378541    4080 flags.go:33] FLAG: --log-backtrace-at=":0"
I0506 23:59:59.378541    4080 flags.go:33] FLAG: --log-dir=""
I0506 23:59:59.378541    4080 flags.go:33] FLAG: --log-file="C:\\etc\\kubernetes\\logs\\kube-proxy.log"
I0506 23:59:59.378541    4080 flags.go:33] FLAG: --log-file-max-size="1800"
I0506 23:59:59.378541    4080 flags.go:33] FLAG: --log-flush-frequency="5s"
I0506 23:59:59.378541    4080 flags.go:33] FLAG: --logtostderr="false"
```

I manually started the kubelet process in powershell on a windows node, and one thing I noticed is that kubelet did log all the flags at the beginning, but after that, it output EVERYTHING to stdout.
This means that the log settings were somehow overridden afterwards. Then I tried disabling the kubelet config by setting `--config=""`, and kubelet generated limited output to stdout/stderr, and logged the rest of the messages to the file.

terminal output:
```
Flag --serialize-image-pulls has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Flag --max-pods has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Flag --pod-manifest-path has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Flag --enable-debugging-handlers has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Flag --resolv-conf has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Flag --cgroups-per-qos has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Flag --enforce-node-allocatable has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
E0507 00:16:37.819918    3588 server.go:692] Kubelet needs to run as uid `0`. It is being run as -1
E0507 00:16:37.821909    3588 file_unsupported.go:29] Watching source file is unsupported in this build
E0507 00:16:37.916609    3588 aws_credentials.go:77] while getting AWS credentials NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors
E0507 00:16:38.069955    3588 cni.go:331] Error adding default_abcdef/47e809742892e3b194fe134d01d269f556e6aa7b2deddb0d1abfd61c55e701eb to network win-bridge/l2bridge: error while ProvisionEndpoint(984ca6ae5db87cfe9b74b7d9e1b0f96ca666966e4db67b09078f5135938cc338_l2bridge,05A966A4-C976-4D1B-BBFD-5BCE782DEB7E,47e809742892e3b194fe134d01d269f556e6aa7b2deddb0d1abfd61c55e701eb): failed to find HNSEndpoint 984ca6ae5db87cfe9b74b7d9e1b0f96ca666966e4db67b09078f5135938cc338_l2bridge: Endpoint 984ca6ae5db87cfe9b74b7d9e1b0f96ca666966e4db67b09078f5135938cc338_l2bridge not found
E0507 00:16:38.072878    3588 cni_windows.go:50] error while adding to cni network: error while ProvisionEndpoint(984ca6ae5db87cfe9b74b7d9e1b0f96ca666966e4db67b09078f5135938cc338_l2bridge,05A966A4-C976-4D1B-BBFD-5BCE782DEB7E,47e809742892e3b194fe134d01d269f556e6aa7b2deddb0d1abfd61c55e701eb): failed to find HNSEndpoint 984ca6ae5db87cfe9b74b7d9e1b0f96ca666966e4db67b09078f5135938cc338_l2bridge: Endpoint 984ca6ae5db87cfe9b74b7d9e1b0f96ca666966e4db67b09078f5135938cc338_l2bridge not found
```

**Comment by mtaufen** [comment]
Likely fix: https://github.com/kubernetes/kubernetes/pull/77562

**Comment by mtaufen** [comment]
Official fix by updating klog to v0.3.1: https://github.com/kubernetes/kubernetes/pull/78216
